subsys/softdevice_handler/nrf_sdh.c

```diff
@@ -14,11 +14,25 @@
 
 LOG_MODULE_REGISTER(nrf_sdh, CONFIG_NRF_SDH_LOG_LEVEL);
 
-#if ((CONFIG_NRF_SDH_CLOCK_LF_SRC == NRF_CLOCK_LF_SRC_RC) &&                                       \
-	(CONFIG_NRF_SDH_CLOCK_LF_ACCURACY != NRF_CLOCK_LF_ACCURACY_500_PPM))
-#warning Please select NRF_CLOCK_LF_ACCURACY_500_PPM when using NRF_CLOCK_LF_SRC_RC
+#if (CONFIG_NRF_SDH_CLOCK_LF_SRC == NRF_CLOCK_LF_SRC_XTAL)
+
+BUILD_ASSERT(CONFIG_NRF_SDH_CLOCK_LF_RC_CTIV == 0, "rc_ctiv must be 0 when using LFXO");
+BUILD_ASSERT(CONFIG_NRF_SDH_CLOCK_LF_RC_TEMP_CTIV == 0, "rc_temp_ctiv must be 0 when usings LFXO");
+BUILD_ASSERT(CONFIG_NRF_SDH_CLOCK_LF_ACCURACY == 0, "accuracy must be 0 when using LFXO");
+
+#if defined(CONFIG_NRF_GRTC_START_SYSCOUNTER) && !defined(CONFIG_NRF_GRTC_TIMER_SOURCE_LFXO)
```

Comment:
Regarding SYSCOUNTER, should we add a separate WARNING ?
SoftDevice (MPSL) have the following requirements: Before enabling SoftDevice: "GRTC must be started by the application ......... the SYSCOUNTER must be enabled ......."
ref: https://docs.nordicsemi.com/bundle/ncs-latest/page/nrfxlib/mpsl/doc/mpsl.html#:~:text=For%20the%20nRF54L%20Series%3A

Reply:
Added a build assert. Please have a look.